core/src/main/scala/org/apache/spark/scheduler/EventLoggingListener.scala

```diff
@@ -132,7 +132,9 @@ private[spark] class EventLoggingListener(
     writer.foreach(_.println(compact(render(eventJson))))
     if (flushLogger) {
       writer.foreach(_.flush())
-      hadoopDataStream.foreach(hadoopFlushMethod.invoke(_))
+      if (!Utils.inShutdown()) {
```

Comment:
A recent discussion noted that this method is kind of hacky. Maybe it's necessary. But is there really no way to test whether the `FileSystem` is closed directly? Really this could be avoided in all eventualities if these operations aren't invoked when the FS is closed, even outside shutdown. + @sryza who might know the API better than I